BuilderES6.js

```diff
@@ -0,0 +1,40 @@
+export default class BuilderES6 {
+  constructor (initialValue) {
+    if (this.constructor === BuilderES6) {
```

Comment:
Adding the abstract class guard is out of scope of this task :)

Reply:
Yeah, but it wasn't forbidden to make it either :)